tools/npm/npm-guide.md

```diff
@@ -0,0 +1,183 @@
+
+### What is npm?
+
+**npm** stands for **Node Package Manager** and helps with installation and management of packages in Node.js by providing a useful command line interface.
+
+### What is a Node package?
+
+A Node package is a library of code that implements a functionality. Naturally, developers want to import and use it within a project instead of re writing it.
+
+### Where can i get the Node packages from?
+
+Users open source and publish their packages in [npm Registry](https://www.npmjs.com/search). Since these packages are open and free, developers usually check README, stats and stars of each package in [GitHub](https://www.github.com) in order to decide how useful is a library before using it.
+
+### Installing npm
+
+npm is installed along with [NodeJS](https://nodejs.org/en/download/) installation. This link provides NodeJS binaries and installers.
+
+Alternatively for linux users run the following command from terminal replacing the node version `setup_8.x`. This example is for node version 8.
+```sh
+curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+sudo apt-get install -y nodejs
+```
+
+# package.json
+
+package.json file is created automatically from npm when initializing a project. It holds information for identifying and managing a project. It looks like that:
+```js
+{
+  "name": "my_app",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "react": "^15.5.4"
+  },
+  "devDependencies": {
+    "eslint": "^3.19.0",
+   }
+}
+```
+
+It holds the list of all dependencies and devDependencies packages. Dependencies are the modules needed to be installed for a project to run while devDependencies are the modules used by the developer to make his life easier. 
+
+Versions of Node packages in package.json file follow the [semantic versioning rules](http://semver.org/). They can also contain the caret(^) or tilde(~) character in front of them. Caret instructs npm to install up to the latest available minor version, while tilde up to the latest available patch version.
+
+When `npm install` runs all packages are installed from both "dependencies" and "devDependencies" sections.
+
+# Commands
+
+### Check version
+
+**npm** version
+
+```sh
+npm -v
+```
+
+**node** version
+
+```sh
+node -v
+```
+
+### Initialialize project
+
+Starts a process that will create **package.json** file in your current folder. While initializing a questionary command line will be initiated where at least **name** and **version** of the project should be provided.
+
+```sh
+npm init
+```
+
+### Installation
+
+Install all project packages as listed in package.json file locally.
+
+```sh
+npm install
+```
+
+Install single package locally.
+
+```sh
+npm install <package_name>
+```
+
+Install single package globally. That bein said, packages installed that way be run from any path.
```

Comment:
We need to explain that this should generally be avoided and only utils packages should be installed globally.